### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Bekovmi**
Increase speed of training

### CODE REVIEW

File: neuro/scripts/download_dataset.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+download-gdrive 1cD0euGOhuOm8w-XXa7b4wQeZ_RfkmxN4 brain_data.zip
+mkdir data
+mkdir data/Mindbonggle_101
+unzip brain_data.zip &>/dev/null
+mv brain_data/* data/Mindbonggle_101
+rm brain_data.zip
```
Comment (Scitator) [inline neuro/scripts/download_dataset.sh]: remove please

File: training/reader.py [hunk 2]
```diff
@@ -92,10 +70,10 @@ def __call__(self, element):
         Returns:
             np.ndarray: Image
         """
-        coords = ast.literal_eval(element[self.coords])
-        subvolume_shape = np.array([64, 64, 64])
         image_name = str(element[self.input_key])
-        with open("./presets/label_protocol_unique.txt", "r") as f:
+        with open(
+            "/home/Bekovmi/neuro/presets/label_protocol_unique.txt", "r"
```
Comment (Scitator) [inline training/reader.py]: return default name please

File: training/runner.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    SupervisedExperiment,
+    utils,
+)
+from catalyst.dl.runner import Runner
+from catalyst.utils.tools.typing import Device, Model
+
+logger = logging.getLogger(__name__)
+
+
+class NeuroRunner(Runner):
```
Comment (Scitator) [inline training/runner.py:15]: are you sure you need whole Runner copy-pasted? what would you like to do?

**Review by Scitator — CHANGES_REQUESTED** [review]

File: training/runner.py (hunk 3 above)
Comment (Bekovmi) [reply training/runner.py:15]: Some classes are abstract, I couldn't inherited

Comment (Scitator) [reply training/runner.py:15]: could you add some examples of `abstract` classes?

**Comment by Scitator** [comment]
btw, @Bekovmi could you please fix codestyle?
you can use https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md#codestyle for more info

File: training/runner.py (hunk 3 above)
Comment (Bekovmi) [reply training/runner.py:15]: I mix up classes with methods. If I rewrite following methods: _run_batch, _run_loader and add _crop_data, I will receive error with inherited _handle_batch.

Comment (Scitator) [reply training/runner.py:15]: what could not you just rewrite `_handle_batch`?

Comment (Scitator) [reply training/runner.py:15]: what functionality have you rewrite?

Comment (Bekovmi) [reply training/runner.py:15]: new _run_batch with _run_loader

Comment (Scitator) [reply training/runner.py:15]: then... why have you copy-paste whole runner? could be redefine only really required parts?

Comment (Bekovmi) [reply training/runner.py:15]: Потому что если полностью не скопировать, возникает ошибка.

Comment (Scitator) [reply training/runner.py:15]: what error? have you tried to debug?

File: training/runner.py [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+            # when model returns dict
+            self._process_output = self._process_output_none
+        else:
+            raise NotImplementedError()
+
+    def _init(self):
+        self.experiment: SupervisedExperiment = None
+        self.state: State = None
+
+    def _crop_data(self, batch, coords):
```
Comment (Scitator) [inline training/runner.py:81]: could you remove this function to dataset?

**Review by Scitator — CHANGES_REQUESTED** [review]

**Pull request closed without merging**